### PR TITLE
unexporting buffer in view resolves #88

### LINKED ISSUE #88: Stop exporting buffer in view
I suggest stop exporting [buffer in view](https://github.com/limetext/lime-backend/blob/7392b2626938bfb62808c8d7ce4b5e7b463fd21c/lib/view.go#L350), we should just export some necessary functionality through view e.g functions currently included in [view_buffer_generated.go](https://github.com/limetext/lime-backend/blob/7392b2626938bfb62808c8d7ce4b5e7b463fd21c/lib/sublime/api/view_buffer_generated.go) so we will have `Fullline`, `Line`, `Substr`, ... as separate methods for `View`. 

My main reasons are:

- There is no need for other packages to access whole buffer functionality through view
- Also we are making some changes directly to buffer like `v.Buffer().SetFileName(filename)` [here](https://github.com/limetext/lime-backend/blob/7392b2626938bfb62808c8d7ce4b5e7b463fd21c/lib/window.go#L79) that should be STH like `v.SetFileName(filename)` so we can manage it from view

### PR DESCRIPTION
**Pull request by zoli**

### CODE REVIEW

**Comment by zoli** [comment]
After merging this the diff on #90 will be much less.

**Comment by zoli** [comment]
This will break frontends because they use view.Buffer directly.

File: lib/view.go [hunk 1]
```diff
@@ -1000,3 +993,79 @@ func (v *View) EraseStatus(key string) {
...
+
+func (v *View) Word(off int) Region {
+	return v.buffer.Word(off)
+}
+
+func (v *View) WordR(r Region) Region {
+	return v.buffer.WordR(r)
+}
+
+func (v *View) AddObserver(ob BufferObserver) error {
```
Comment (ricochet1k) [inline lib/view.go]: Am I mistaken in thinking that this BufferObserver is now the only way for a frontend to get access to the Buffer through a View? Would it make sense to create a ViewObserver instead?

**Comment by ricochet1k** [comment]
Is there a way to create multiple Views for one Buffer?

File: lib/view.go (hunk 1 above)
Comment (zoli) [inline lib/view.go]: Yes this is the only way currently, Could you add more details on ViewObserver.
Also we have `OnModified` event which I think its a bit inconsistent with the observers.

**Comment by zoli** [comment]
> Is there a way to create multiple Views for one Buffer?

currently no but its on the features list that we would like to have, I doubt that sublime text has this feature.

**Comment by zoli** [comment]
Some notes on this pr for later:
I know that writing code like this isn't right, It's repetative and inefficient.
```go
func (v *View) RowCol(point int) (int, int) {
	return v.buffer.RowCol(point)
}

func (v *View) TextPoint(row, col int) int {
	return v.buffer.TextPoint(row, col)
}
```

The best solution would be defining restricted buffer interface in backend package and embedding it in View type but then there would be some problems with [composite actions](https://github.com/limetext/text/blob/master/action.go) which we are using in Edit type and maybe some other issues so until then I think the best way to go is what I did in 699c318.

**Comment by ricochet1k** [comment]
If there are plans to have multiple views for one buffer, then the backend should expose a Buffer interface, probably separate from the one in the `text` package. If we don't want to have multiple views on the same buffer, then probably the best way is to turn the backend View into a BufferObserver and forward modifications to all registered ViewObservers that just forwards all modified events through with the view object instead of the buffer object. Either way, it is pretty useless to get a Buffer in the BufferObserver interface without being able to map it back to the View. IIRC, right now, the only BufferObserver is the frontend's view.

**Pull request merged**